torch/csrc/jit/runtime/profiling_graph_executor_impl.cpp

```diff
@@ -116,6 +118,59 @@ static bool needsGradientInProfilingMode(Block* b) {
   return false;
 }
 
+bool guardDifferentiableGraph(Node* dnode) {
+  auto gi = dnode->g(attr::Subgraph)->inputs();
+  bool all_inputs_seen = true;
+  for (size_t i = 0; i < gi.size(); i++) {
+    auto ty = gi[i]->type()->cast<TensorType>();
+    if (ty) {
+      auto n = gi[i]->uses().at(0).user;
```

Comment:
I suppose with DifferentiableGraph's we're always fusing just a block, so all or no uses will have been executed